PowerEditor/src/WinControls/ProjectPanel/TreeView.cpp

```diff
@@ -427,6 +453,9 @@ bool TreeView::swapTreeViewItem(HTREEITEM itemGoDown, HTREEITEM itemGoUp)
 	tvInsertUp.hInsertAfter = itemTop;
 	tvInsertUp.hParent = parentGoUp;
 	HTREEITEM hTreeParent1stInserted = (HTREEITEM)::SendMessage(_hSelf, TVM_INSERTITEM, 0, (LPARAM)(LPTVINSERTSTRUCT)&tvInsertUp);
+	_validHandles.insert(hTreeParent1stInserted);

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `NULL` before using `newItem`.

***Especially*** since you're putting it in a variable named `_validHandles`.